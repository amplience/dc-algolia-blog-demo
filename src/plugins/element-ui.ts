import Vue from 'vue';
import { Card, Row, Col, Container, Header, Main, Pagination } from 'element-ui';

export default (): void => {
  Vue.use(Card);
  Vue.use(Row);
  Vue.use(Col);
  Vue.use(Container);
  Vue.use(Header);
  Vue.use(Main);
  Vue.use(Pagination);
};
