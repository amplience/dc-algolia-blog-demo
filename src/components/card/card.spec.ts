import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { Card as ElCard } from 'element-ui';
import Card from './card.vue';

describe('Card', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Card, {
      stubs: { ElCard, NuxtLink: RouterLinkStub },
      filters: {
        date: value => value
      },
      propsData: {
        title: 'test-title',
        description: 'test-description',
        link: 'test-id',
        timestamp: '2011-11-18T14:54:39.929Z'
      }
    });

    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
