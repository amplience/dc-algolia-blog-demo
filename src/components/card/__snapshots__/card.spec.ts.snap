// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is a Card instance with card data 1`] = `
<div
  class="el-card is-hover-shadow"
>
  <!---->
  <div
    class="el-card__body"
    style="padding: 0px;"
  >
    <div>
      <a>
        <img
          alt="test-title"
          class="image"
          src="test-image.jpg"
        />
         
        <div
          class="content"
        >
          <h2>
            test-title
          </h2>
           
          <time
            class="date-time"
            datetime="datetime-local"
          >
            2011-11-18 14:54:39
          </time>
        </div>
      </a>
    </div>
  </div>
</div>
`;
